modules/fbs-core/math-parser/src/main/kotlin/de/thm/ii/fbs/mathParser/transformers/rules/OperationRule.kt

```diff
@@ -0,0 +1,12 @@
+package de.thm.ii.fbs.mathParser.transformers.rules
+
+import de.thm.ii.fbs.mathParser.ast.Expr
+import de.thm.ii.fbs.mathParser.ast.Operation
+
+abstract class OperationRule : Rule {
+    abstract fun matchesOperation(operation: Operation): Boolean
+    abstract fun applyOperation(operation: Operation): Expr
+
+    override fun matches(expr: Expr): Boolean = expr is Operation && matchesOperation(expr)
```

Comment:
These line is not fully coverd by the tests (see [codecov](https://app.codecov.io/gh/thm-mni-ii/feedbacksystem/pull/1165#f6cc104337108fe27760b9d594216833-R10))